lib/adagrams.rb

```diff
@@ -0,0 +1,136 @@
+require 'pry'
+
+#create data structure
+all_letters = []
+
+letters = {"A" => 9, "B" => 2, "C" => 2, "D"=> 4, "E" => 12, "F"=> 2, "G"=>3, "H"=>2, "I" => 9, "J"=>1, "K"=> 1, "L"=>4, "M"=> 2, "N"=> 6, "O"=>8, "P" => 2, "Q" => 1, "R"=>6, "S"=>4, "T"=>6, "U"=>4, "V"=>2, "W"=>2, "X"=>1, "Y"=>2, "Z"=>1}
+
+
+# Fill pool of letters array with letters
+letters.each do |letter, freq|
+  count = 0
+  until count == freq
+    all_letters << letter
+    count += 1
+  end
+end
+
+
+# Define method for creating user hand
+def draw_letters(array)
+  user_letters = array.sample(10)
+  return user_letters
+end
+
+DRAWN_LETTERS = draw_letters(all_letters)
+
+
+# Transform input word string into array, all in capital letters
+def make_word_array(input)
+  input = input.upcase.split("")
+  return input
+end
+
+
+#create method to see if word uses letters that are actually available
+def uses_available_letters? (input, drawn_letters)
+  is_valid = "default"
+  word_letters = make_word_array(input)
+  word_letters.each do |letter|
```

Comment:
This method is destructive! Because you call `delete_at` on the `drawn_letters` directly, the hand will be reduced by the letters in the input. For example:

```ruby
hand = draw_letters
puts hand
# => ["T", "T", "E", "F", "R", "A", "E", "L", "V", "A"]
uses_available_letters?('tte', hand)
puts hand
# => ["F", "R", "A", "E", "L", "V", "A"]
```

While none of the tests we provided check for destroying the hand, this is bad because it's unexpected behavior. Nothing about "check whether this word is made up of letters in this hand" suggests that the hand will be changed in the process, but that's exactly what happens.

The way to address this problem would be to make a copy of the hand before removing letters from it.